src/inference/hmc.jl

```diff
@@ -663,16 +504,100 @@ observe(spl::Sampler{<:Hamiltonian},
 #### Default HMC stepsize and mass matrix adaptor
 ####
 
-function AHMCAdaptor(alg::AdaptiveHamiltonian; init_ϵ=alg.init_ϵ)
-    p = AHMC.Preconditioner(getmetricT(alg))
-    nda = AHMC.NesterovDualAveraging(alg.δ, init_ϵ)
+function AHMCAdaptor(alg::AdaptiveHamiltonian; ϵ=alg.ϵ)
```

Comment:
I think this function can be easily refactored using multiple dispatch now. Shall we do this quickly?

Reply:
Dispatching on what? The `alg` type? Wouldn't we need to catch four combinations (HMCDA/NUTS and `UnitEuclidian`/`Other`)?

Reply:
You're right. Let's not add this complexity for now.

Reply:
As you pointed out somewhere else, we can improve the way users set adaptor as well. But this should be in another PR.